catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/CachedSubscription.java

```diff
@@ -0,0 +1,184 @@
+            throw new SubscriptionRegistrationException(format(
+                    "Factory type mismatch for subscription type [%s]",
+                    metadata.getTypeName()));
+        }
+
+        LOGGER.debug("Regenerating subscription [ {} | {} | {} ]",
+                metadata.getId(),
+                metadata.getTypeName(),
+                metadata.getCallbackAddress());
+
+        Subscription subscription = factory.createSubscription(metadata);
+        if (subscription == null) {
+            throw new SubscriptionRegistrationException("Regenerated subscription cannot be null");
+        }
+        registerSubscription(subscription);
+    }
+
+    /**
+     * Register the provided {@link Subscription} as this cached subscription's OSGi service.
+     *
+     * @param subscription the subscription to register.
+     * @throws SubscriptionRegistrationException if registration failed.
+     */
+    public void registerSubscription(Subscription subscription) {
+        LOGGER.debug("Registering service [{}]", metadata.getId());
+
+        Dictionary<String, String> properties = new Hashtable<>();
+        properties.put(SUBSCRIPTION_ID_OSGI, metadata.getId());
+        properties.put(EVENT_ENDPOINT, metadata.getCallbackAddress());
+
+        registration = getBundleContext().registerService(Subscription.class.getName(),
+                subscription,
+                properties);
+
+        if (registration != null) {
+            LOGGER.debug("Subscription [ {} | {} | {} ] registered with bundle ID = {} ",
+                    metadata.getId(),
+                    metadata.getTypeName(),
+                    metadata.getCallbackAddress(),
+                    registration.getReference()
+                            .getBundle()
+                            .getBundleId());
+            this.subscription = subscription;
+        } else {
+            throw new SubscriptionRegistrationException(format(
+                    "Subscription registration attempt for id [%s] of type [%s] was null",
+                    metadata.getId(),
+                    metadata.getTypeName()));
+        }
+    }
+
+    /**
+     * Removes this cached subscriptions OSGi service from the service registry. Calling this method
+     * on a cached subscription that {@link #isNotRegistered()} will throw an exception.
+     *
+     * @throws SubscriptionRegistrationException if this cached subscription did not have a service
+     *                                           registered.
+     */
+    public void unregisterSubscription() {
+        LOGGER.debug("Removing service [{}]", metadata.getId());
+        if (registration == null) {
+            throw new SubscriptionRegistrationException(format(
+                    "Subscription [%s] had no registration and could not be unregistered. ",
+                    metadata.getId()));
+        }
+        registration.unregister();
+        registration = null;
```

Comment:
This opens the door to a lot of potential issues. `CachedSubscriptions` are not meant to be reused. You pass in the metadata object up front, but can't change it after. Effectively, a cached subscription only tracks a single subscription's "lifecycle" through osgi. 

Resetting the subscription to `null` also means we can't return the object after a delete operation. This change will cause an NPE in the next PR.